src/components/input/StateEditor.ts

```diff
@@ -0,0 +1,69 @@
+import { State } from "./State"
+
+export class StateEditor {
+	get value(): string { return this.state.value }
+	private get selectionStart(): number { return this.state.selectionStart }
+	private set selectionStart(selectionStart: number) { this.state.selectionStart = selectionStart }
+	private get selectionEnd(): number { return this.state.selectionEnd }
+	private set selectionEnd(selectionEnd: number) { this.state.selectionEnd = selectionEnd }
+
+	private constructor(private state: State) { }
+
+	get(index: number): string {
+		return this.value[index]
+	}
+	isDigit(index: number): boolean {
+		const character = this.get(index)
+		return character >= "0" && character <= "9"
+	}
+	toState(): State {return { value: this.state.value, selectionStart: this.selectionStart, selectionEnd: this.selectionEnd}}
```

Comment:
Move this method too just above the static methods. That way we first have the predicates and then the modifiers.